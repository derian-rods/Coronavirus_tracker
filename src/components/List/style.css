.app__list{
    flex-direction: column;
    flex-wrap: nowrap;
    width:40vw;
    margin: auto;
    padding-top: 3em;
    padding-left: 3em;
}


.list{
    margin: 0em 1em;
    border-top: 5px solid#ff9710;
    padding: 1em;
    border-radius: 10px;
    background: #FAFAFB;
    padding: 2em;
    width: 68%;
    overflow-y: auto;
    height: 41em;
    box-shadow: 0 10px 10px #00000086;
}
.list__title{
    padding: 1em 0em;
}
ul{
    list-style: none;
}
li{
    display: flex;
    align-items: center;
    justify-content: space-between;
    border: solid 2px #6f6466;
    padding: 0 1em; 
    height: 3em;
}

li:nth-child(2n+1){
    background: #FF9710;
}
.list__flag{
    width: 6%;
    margin-right: .4rem;
}
.list__cases{
    color: #9b7076;
    margin-left: .5rem;
}

/* @media screen and (max-width: 1024px){
   .app__list{
        width:34vw;
        padding-left: 0;
    }
    .list{
        margin: -2em 1em 0em 5em;
        border-top: 5px solid#ff9710;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 17em;
        overflow-y: auto;
        height: 39em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 14em;
        padding: 1em;
        margin-left: 1em;
        height: 1em;
    }
    li div{
        display: flex;
    }
    .list__title span{
        margin-top: ;
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
} */

@media screen and (max-width: 1024px){
    .app__list{
        width:20em;
        padding-top: 2em;
        margin: none;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em -15.5em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 40em;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 36em;
        padding: 1em;
        margin-left: 1em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}

@media screen and (max-width: 768px){
    .app__list{
        width:20em;
        padding-top: 2em;
        margin: none;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em -15.5em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 40em;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 36em;
        padding: 1em;
        margin-left: 1em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}


/* 414 */
@media screen and (max-width: 414px){
    .app__list{
        width:100vw;
        padding-top: 2em;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em -5.5em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 17em;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 13em;
        padding: 1em;
        margin-left: 1em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}


@media screen and (max-width: 411px){
    .app__list{
        width:100vw;
        padding-top: 2em;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em 2.5em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 17em;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 13em;
        padding: 1em;
        margin-left: 1em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}
/* 384 */
@media screen and (max-width: 384px){
    .app__list{
        width:100vw;
        padding-top: 2em;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em 1.8em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 17em;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 16.5em;
        padding: 1em;
        margin-left: -.8em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}
/* 375 */
@media screen and (max-width: 375px){
    .app__list{
        width:100vw;
        padding-top: 2em;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em 2.5em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 60%;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 13em;
        padding: 1em;
        margin-left: -.8em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}
@media screen and (max-width: 360px){
    .app__list{
        width:100vw;
        padding-top: 2em;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em 2.5em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 60%;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 13em;
        padding: 1em;
        margin-left: -.8em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}
@media screen and (max-width: 320px){
    .app__list{
        width:100vw;
        padding-top: 2em;
        padding-left: 0;
    }
    .list{
        margin: 0em 1em 0em 2em;
        border-top: 5px solid#ff9710;
        padding: 1em;
        border-radius: 10px;
        background: #FAFAFB;
        padding: 2em;
        width: 12em;
        overflow-y: auto;
        height: 25em;
        box-shadow: 0 10px 10px #00000086;
        margin-bottom: 2em;
    }
    li{
        width: 12em;
        padding: 1em;
        margin-left: -1em;
        height: 1em;
    }
    .list__title{
        padding: -.8em 0em;
    }
    .list__flag{
        width: 15%;
        margin-right: .4rem;
    }
}
