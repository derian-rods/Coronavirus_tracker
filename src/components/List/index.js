import React from 'react'
import PropTypes from 'prop-types'
import './style.css'
export const List = ({countries}) => {
    return (
     <>
                <div className="app__list">
                <div className="list">
                    <h3 className="list__title">Cases By Country</h3>
                    <ul className="list__ul">
                    {
                            countries.map(country => (
                                <li className="list__li" key={country.country}>
                                   <div>
                                       <img className="list__flag" src={country.countryInfo.flag} alt={country.country} />
                                   <span>{country.country}</span>
                                   </div>
                                    <span>{country.cases}<i className="fas fa-virus list__cases"></i></span>
                                </li>
                            ))
                        }
                    </ul>
                </div>
            </div>
     </>
    )
}

    List.propTypes = {
        countries: PropTypes.array.isRequired,
    }