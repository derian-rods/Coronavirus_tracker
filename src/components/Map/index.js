import React, { useEffect, useState } from "react";
import { scaleQuantile } from "d3-scale";
import {Table} from './Table'
import {
  ComposableMap,
  Geographies,
  Geography,
  Sphere,
  ZoomableGroup,
  Graticule,
} from "react-simple-maps";
import PropTypes from 'prop-types'
import './style.css'

const geoUrl =
  "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";

  
  const MapChart = ({countries}) => {
    const [data, setData] = useState([]);
    
    const colorScale = scaleQuantile()
      .domain(data.map(d => d.cases))
      .range([   
        "#ffedea",
        "#ffcec5",
        "#ffad9f",
        "#ff8a75",
        "#ff5533",
        "#e2492d",
        "#be3d26",
        "#9a311f",
        "#782618",
    ]);
      
  useEffect(() => {
    
   const valueInfo = countries.map(country =>({
      name: country.country,
      cases: country.cases,
      iso3: country.countryInfo.iso3,
      deaths: country.deaths,
      recovered: country.recovered
    }));

    setData(valueInfo)
  
  },[countries]);  
  return (
    <div className="app__map">
        <div className="card__map">
          <h2 className="map__title">World Map</h2>
          <div className="map__table">
             <Table />
          </div>
        <ComposableMap
                projectionConfig={{
                    rotate: [-10, 0, 0],
                    scale: 160
                }}
                >
                <ZoomableGroup>
                <Sphere stroke="#E4E5E6" strokeWidth={0.5} />
                <Graticule stroke="#E4E5E6" strokeWidth={0.5} />
                {data.length > 0 && (
                    <Geographies geography={geoUrl}>
                    {({ geographies }) =>
                        geographies.map((geo) => {
                        const d = data.find((s) => s.iso3 === geo.properties.ISO_A3);
                        return (
                            <Geography
                            key={geo.rsmKey}
                            geography={geo}
                            fill={d ? colorScale(d.cases) : "#ffedea"}
                            />
                        );
                        })
                    }
                    </Geographies>
                )}
                </ZoomableGroup>
                </ComposableMap>
        </div>
    </div>
  );
};

MapChart.propTypes = {
  countries: PropTypes.array.isRequired,
}


export default MapChart;