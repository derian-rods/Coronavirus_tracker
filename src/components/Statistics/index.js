 import React from 'react'
import { Card } from '../Layouts/cards';
import PropTypes from 'prop-types'
import './style.css'
 
 export const CardStats = ({data}) => {
     return (
         <div className="app__cardsStats">
             <Card  title='Coronavirus cases' total={data.cases} increment={data.todayCases} icon={'fas fa-viruses'} icon2={'fas fa-angle-double-up'}  color={'redcolor'}/>
             <Card  title='Deaths' total={data.deaths} increment={data.todayDeaths} icon={'fas fa-skull-crossbones'} icon2={'fas fa-angle-double-up'}  color={'redcolor'} />
             <Card title='Recovered' total={data.recovered} increment={data.todayRecovered} icon={'fas fa-ambulance'} icon2={'fas fa-angle-double-up'}  color={'greencolor'}/>
         </div>
     )
 }
 
 CardStats.propTypes = {
     data: PropTypes.object.isRequired
 }