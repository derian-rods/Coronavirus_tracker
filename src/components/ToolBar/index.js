import React from 'react';
import PropTypes from 'prop-types'
import './style.css';

export const ToolBar = ({ countries, country, handleCountryChange}) => {

    return (
        <>
                <div className="navbar">
                <div className="navbar__brand">
                 <img className="brand__img" src='https://www.flaticon.com/svg/static/icons/svg/2785/2785819.svg' alt="corona virus"></img>
                 <span className="brand__text">Tracker</span>
                </div>
                <div className="navbar__card">
                     <select value={country} onChange={handleCountryChange}>
                         <option >World Wide</option>
                         {countries.map(c => (
                             <option key={c.country}>{c.country}</option>
                         ))}
                     </select>
                </div>
             </div>
        </>
    )
}

ToolBar.propTypes = {
    countries: PropTypes.array.isRequired,
    country: PropTypes.string.isRequired,
    hadleCountryChange: PropTypes.func
}