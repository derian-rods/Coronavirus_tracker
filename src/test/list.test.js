import React from 'react';
import '@testing-library/jest-dom';
import {List} from '../components/List';
import {shallow} from 'enzyme';

describe('tests of list component', () => {
    
    test('should show <List /> correctly ', () => {
        const countries = [{
            country: 'Spain', 
            countryInfo:{
                flag: 'https://disease.sh/assets/img/flags/af.png'
            }
        }, 
        {  country: 'EEUU', 
        countryInfo:{
            flag: 'https://disease.sh/assets/img/flags/af.png'
        }},
    ]
        const wrapper = shallow(<List countries={countries} />)
        expect(wrapper).toMatchSnapshot()
    });
})
