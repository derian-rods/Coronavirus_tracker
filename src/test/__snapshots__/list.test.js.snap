// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests of list component should show <List /> correctly  1`] = `
<Fragment>
  <div
    className="app__list"
  >
    <div
      className="list"
    >
      <h3
        className="list__title"
      >
        Cases By Country
      </h3>
      <ul
        className="list__ul"
      >
        <li
          className="list__li"
          key="Spain"
        >
          <div>
            <img
              alt="Spain"
              className="list__flag"
              src="https://disease.sh/assets/img/flags/af.png"
            />
            <span>
              Spain
            </span>
          </div>
          <span>
            <i
              className="fas fa-virus list__cases"
            />
          </span>
        </li>
        <li
          className="list__li"
          key="EEUU"
        >
          <div>
            <img
              alt="EEUU"
              className="list__flag"
              src="https://disease.sh/assets/img/flags/af.png"
            />
            <span>
              EEUU
            </span>
          </div>
          <span>
            <i
              className="fas fa-virus list__cases"
            />
          </span>
        </li>
      </ul>
    </div>
  </div>
</Fragment>
`;
